unit/models/transaction.py

```diff
@@ -463,4 +466,118 @@ def to_dict(self) -> Dict:
             parameters["sort"] = self.sort
         if self.include:
             parameters["include"] = self.include
-        return parameters
\ No newline at end of file
+        return parameters
+
+
+class SimulatePurchaseTransaction(UnitRequest):
+    def __init__(
+        self,
+        amount: int,
+        card_id: str,
+        last_4_Digits: str,
+        deposit_account_id: str,
+        merchantName: str,
+        merchantType: str,
+        merchantLocation: str,
+        direction: str = "Debit",
+        authorization_id: str = None,
+    ):
+        self.authorization_id = authorization_id
+        self.last_4_Digits = last_4_Digits
+        self.deposit_account_id = deposit_account_id
+        self.direction = direction
+        self.amount = amount
+        self.card_id = card_id
+        self.merchantName = merchantName
+        self.merchantType = merchantType
+        self.merchantLocation = merchantLocation
+
+    def to_json_api(self) -> Dict:
+        payload = {
+            "data": {
+                "type": "purchaseTransaction",
+                "attributes": {
+                    "amount": self.amount,
+                    "direction": self.direction,
+                    "last4Digits": self.last_4_Digits,
+                    "merchantName": "The Home Depot",
+                    "merchantType": self.merchantType,
+                    "merchantLocation": self.merchantLocation,
+                    "recurring": False
+                },
+                "relationships": {
+                    "account": {
+                        "data": {
+                            "type": "depositAccount",
```

Comment:
This will be a `creditAccount` now with charge cards

Reply:
Hm, I tested and it seems to work fine with creditAccounts. They may not have updated the api for charge card simulations